configs/example/gem5_library/x86-mi300x-gpu.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2024 Advanced Micro Devices, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Script to run a full system GPU simulation.
+
+Usage:
+------
+```
+scons build/VEGA_X86/gem5.opt
+./build/VEGA_X86/gem5.opt
+    configs/example/gem5_library/x86-viper-gpu.py
+    --image <disk image>
+    --kernel <kernel>
+    --app <gpu application>
```

Comment:
I would prefer to hard-code these and not have command line parameters. We should use things available from gem5-resource (i.e., with `obtain_resource`). I like to think of these files as **examples** for people to model their own scripts off of. I don't think people should use these files directly.

Reply:
What is the alternative since no (non-ancient) GPUFS resources are available online?  I needed a way to be able to test this and gave up trying to make a local json config (got a bunch of unhelpful python errors reading the json). Are the users expected to have to create a gem5-config.json for their local disk images to use obtain_resource?

Reply:
That sounds good. I made the KVM perf change below so if this looks good to you feel free to merge.

Reply:
@Harshil2107: can you put it on your todo list to create and upload these resources ?

Reply:
@Harshil2107  what is the status of this? We can merge this change as-is, but it's a requirement for the release that we update this to use resources instead of this command-line interface.